otel-extension/src/it/scala/io/scalac/mesmer/agent/utils/OtelAgentTest.scala

```diff
@@ -0,0 +1,21 @@
+package io.scalac.mesmer.agent.utils
+
+import io.opentelemetry.context.ContextStorage
+import io.opentelemetry.instrumentation.testing.AgentTestRunner
+import io.opentelemetry.instrumentation.testing.util.ContextStorageCloser
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.TestSuite
+
+abstract class OtelAgentTest extends TestSuite with BeforeAndAfterAll with BeforeAndAfterEach {
+
+  private val testRunner = AgentTestRunner.instance()
```

Comment:
As far as I remember it can use one agent for multiple suites. It does not pick which instrumentations are ones are installed - all are installed at the same time. Imho that's good - in case we have any conflicts/problems due to this, we certainly would have them outside testing. So it's good to detect such scenarios here, if any (didn't spot any problems yet).